components/scream/src/physics/rrtmgp/rrtmgp_utils.hpp

```diff
@@ -43,6 +46,26 @@ namespace scream {
                 return ( (nstep == 0) || (nstep % irad == 0) );
             }
         }
+
+
+        // Verify that array only contains values within valid range, and if not
+        // report min and max of array
+        template <class T> bool check_range(T x, Real xmin, Real xmax, std::string msg) {
+            bool pass = true;
+            auto _xmin = minval(x);
+            auto _xmax = maxval(x);
+            if (_xmin < xmin or _xmax > xmax) {
+                pass = false;
+                std::cout 
+                    << std::string(msg) 
+                    << ": one or more values outside range"
+                    << "; minval = " << _xmin 
+                    << "; maxval = " << _xmax << "\n";
```

Comment:
Shouldn't you also print out what the allowable range was?

Reply:
Done.